src/hera/workflows/runner.py

```diff
@@ -75,14 +75,19 @@ def _parse(value, key, f):
 def _is_str_kwarg_of(key: str, f: Callable):
     """Check if param `key` of function `f` has a type annotation of a subclass of str."""
     type_ = inspect.signature(f).parameters[key].annotation
-    if not type_:
-        return True
-    try:
-        return issubclass(type_, str)
-    except TypeError:
```

Comment:
Confirmed this error is only raised if the type annotation is an object rather than a class

```py
@script()
def object_type_parameter(my_anything: object()) -> Any:
    """Trying to pass an object as the type instead of class to raise TypeError"""
    return my_anything
```

I don't think this really makes sense as Python code so my vote is to remove the TypeError code. Otherwise I can add a function like above and test it